lib/class-facet.php

```diff
@@ -0,0 +1,256 @@
+<?php
+/**
+ * Class for Facets.
+ *
+ * @package Elasticsearch_Extensions
+ */
+
+namespace Elasticsearch_Extensions;
+
+/**
+ * Elasticsearch facet/aggregation.
+ */
+class Facet {
+	/**
+	 * The label for this facet, as provided by ES.
+	 *
+	 * @var string
+	 */
+	public string $label;
+
+	/**
+	 * This facet's buckets (results).
+	 *
+	 * @var array
+	 */
+	public array $buckets;
+
+	/**
+	 * Facet Config.
+	 *
+	 * @var array
+	 */
+	public array $config;
+
+	/**
+	 * The human-readable name for this facet section.
+	 *
+	 * @var string
+	 */
+	public string $name;
+
+	/**
+	 * The label for this facet section.
+	 *
+	 * @var string
+	 */
+	public string $title;
+
+	/**
+	 * The parsed type from the facet label.
+	 *
+	 * @var string
+	 */
+	public string $type;
+
+	/**
+	 * The parsed subtype from the facet label, if applicable.
+	 *
+	 * @var string
+	 */
+	public string $subtype;
```

Comment:
Nitpicky, but I like to alphabetize my class properties and method names to keep everything organized.